bundle/src/main/java/com/adobe/acs/commons/models/injectors/impl/ParentResourceValueMapValueInjector.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2018 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.models.injectors.impl;
+
+import com.adobe.acs.commons.models.injectors.annotation.ParentResourceValueMapValue;
+import com.day.cq.wcm.api.NameConstants;
+import com.day.cq.wcm.foundation.forms.MergedValueMap;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.spi.DisposalCallbackRegistry;
+import org.apache.sling.models.spi.Injector;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Injects the parent component property
+ */
+@Component(property = Constants.SERVICE_RANKING + ":Integer=" + Integer.MAX_VALUE, service = Injector.class)
+public class ParentResourceValueMapValueInjector implements Injector {
+
+    @Override
+    public String getName() {
+        return ParentResourceValueMapValue.SOURCE;
+    }
+
+    @Override
+    public Object getValue(Object adaptable, String name, Type type, AnnotatedElement annotatedElement, DisposalCallbackRegistry disposalCallbackRegistry) {
+        Integer maxLevel = annotatedElement.getAnnotation(ParentResourceValueMapValue.class).maxLevel();
```

Comment:
Should probably check the annotation, similar to what's being done in com.adobe.acs.commons.models.injectors.impl.HierarchicalPagePropertyInjector#getValue

``` 
       if (!element.isAnnotationPresent(HierarchicalPageProperty.class)) {
            //skipping javax.Inject for performance reasons. Only supports direct injection.
            return null;
        }
```

Reply:
Sure. Will add the check.